### PR TITLE
Add a column in aws_redshift_cluster table to store audit logging information. Closes #333

### LINKED ISSUE #333: Add a column in `aws_redshift_cluster` table to store  audit logging information
**Is your feature request related to a problem? Please describe.**

Need to validate below use case but didn't find any column storing the value for 'Audit Loggin`

use case: Check if Redshift cluster has audit logging enabled

**Describe the solution you'd like**
Need a column to store this information.

**Additional context**
image
image

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_redshift_cluster []

PRETEST: tests/aws_redshift_cluster

TEST: tests/aws_redshift_cluster
Running terraform
aws_vpc.my_vpc: Creating...
aws_vpc.my_vpc: Still creating... [10s elapsed]
aws_vpc.my_vpc: Creation complete after 19s [id=vpc-03469fa3c39de898d]
aws_internet_gateway.igw: Creating...
aws_internet_gateway.igw: Creation complete after 6s [id=igw-0c8027f1e6942a06d]
aws_subnet.my_subnet1: Creating...
aws_subnet.my_subnet2: Creating...
aws_subnet.my_subnet2: Creation complete after 4s [id=subnet-05121773bfed20728]
aws_subnet.my_subnet1: Creation complete after 5s [id=subnet-04320b80dd353904b]
aws_redshift_subnet_group.my_subnet_group: Creating...
aws_redshift_subnet_group.my_subnet_group: Creation complete after 3s [id=turbottest32649]
aws_redshift_cluster.named_test_resource: Creating...
aws_redshift_cluster.named_test_resource: Still creating... [10s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [20s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [30s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [40s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [50s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [1m0s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [1m10s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [1m20s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [1m30s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [1m40s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [1m50s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [2m0s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [2m10s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [2m20s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [2m30s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [2m40s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [2m50s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [3m0s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [3m10s elapsed]
aws_redshift_cluster.named_test_resource: Still creating... [3m20s elapsed]
aws_redshift_cluster.named_test_resource: Creation complete after 3m21s [id=turbottest32649]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Warning: Interpolation-only expressions are deprecated

  on variables.tf line 78, in resource "aws_redshift_subnet_group" "my_subnet_group":
  78:   subnet_ids = ["${aws_subnet.my_subnet1.id}", "${aws_subnet.my_subnet2.id}"]

Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.

Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.

(and one more similar warning elsewhere)


Apply complete! Resources: 6 added, 0 changed, 0 destroyed.

Outputs:

account_id = "013122550996"
aws_partition = "aws"
region_name = "us-east-1"
resource_aka = "arn:aws:redshift:us-east-1:013122550996:cluster:turbottest32649"
resource_name = "turbottest32649"

Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:013122550996:cluster:turbottest32649"
    ],
    "allow_version_upgrade": true,
    "cluster_identifier": "turbottest32649"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:013122550996:cluster:turbottest32649"
    ],
    "allow_version_upgrade": true,
    "cluster_identifier": "turbottest32649"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:013122550996:cluster:turbottest32649"
    ],
    "cluster_identifier": "turbottest32649"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:013122550996:cluster:turbottest32649"
    ],
    "cluster_identifier": "turbottest32649",
    "tags": {
      "name": "turbottest32649"
    },
    "title": "turbottest32649"
  }
]
✔ PASSED

POSTTEST: tests/aws_redshift_cluster

TEARDOWN: tests/aws_redshift_cluster

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
 select
  cluster_identifier,
  logging_status -> 'LoggingEnabled' as LoggingEnabled
from
  aws_redshift_cluster where cluster_identifier = 'redshift-cluster-1'
+--------------------+----------------+
| cluster_identifier | loggingenabled |
+--------------------+----------------+
| redshift-cluster-1 | false          |
+--------------------+----------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_redshift_cluster.go [hunk 1]
```diff
@@ -373,6 +380,39 @@ func getRedshiftCluster(ctx context.Context, d *plugin.QueryData, h *plugin.Hydr
...
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	var name string
+	if h.Item != nil {
+		name = *h.Item.(*redshift.Cluster).ClusterIdentifier
+	} else {
+		name = d.KeyColumnQuals["cluster_identifier"].GetStringValue()
+	}
```
Comment (Subhajit97) [inline aws/table_aws_redshift_cluster.go:396]: Why do we need this block for a hydrate function?

File: docs/tables/aws_redshift_cluster.md [hunk 2]
```diff
@@ -60,3 +57,13 @@ from
...
 ```
+
+### List clusters audit logging status
+
+```sql
+select
+  cluster_identifier,
+  logging_details -> 'LoggingEnabled' as LoggingEnabled
+from
+  aws_redshift_cluster;
```
Comment (Subhajit97) [inline docs/tables/aws_redshift_cluster.md:68]: Can we update this example to list logging status for a specific cluster, instead of listing all clusters?

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav There are few changes required. Please see comments, thanks!

File: aws/table_aws_redshift_cluster.go (hunk 1 above)
Comment (bigdatasourav) [reply aws/table_aws_redshift_cluster.go:396]: yes , we dont need to fetch it from cluster_identifier.

File: aws/table_aws_redshift_cluster.go [hunk 3]
```diff
@@ -373,6 +380,34 @@ func getRedshiftCluster(ctx context.Context, d *plugin.QueryData, h *plugin.Hydr
...
 
+func getRedshiftLoggingDetails(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// TODO Put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	name := *h.Item.(*redshift.Cluster).ClusterIdentifier
```
Comment (Subhajit97) [inline aws/table_aws_redshift_cluster.go:391]:
```suggestion
	name := h.Item.(*redshift.Cluster).ClusterIdentifier
```

File: aws/table_aws_redshift_cluster.go [hunk 4]
```diff
@@ -373,6 +380,34 @@ func getRedshiftCluster(ctx context.Context, d *plugin.QueryData, h *plugin.Hydr
...
+
+	// Create service
+	svc, err := RedshiftService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	params := &redshift.DescribeLoggingStatusInput{
+		ClusterIdentifier: aws.String(name),
```
Comment (Subhajit97) [inline aws/table_aws_redshift_cluster.go:400]:
```suggestion
		ClusterIdentifier: name,
```

File: aws/table_aws_redshift_cluster.go [hunk 5]
```diff
@@ -270,6 +264,19 @@ func tableAwsRedshiftCluster(_ context.Context) *plugin.Table {
 				Description: "A list of Amazon Virtual Private Cloud (Amazon VPC) security groups that are associated with the cluster. This parameter is returned only if the cluster is in a VPC.",
 				Type:        proto.ColumnType_JSON,
 			},
+			{
+				Name:        "logging_details",
```
Comment (cbruno10) [inline aws/table_aws_redshift_cluster.go:268]: Should this column be called `logging_status` instead? The API call is `DescribeLoggingStatus, so it seems more intuitive to follow with the column name.

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see comment

File: aws/table_aws_redshift_cluster.go (hunk 5 above)
Comment (bigdatasourav) [reply aws/table_aws_redshift_cluster.go:268]: updated

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**